class-components/src/components/CardList/CardList.test.tsx

```diff
@@ -0,0 +1,49 @@
+import { describe, it, expect } from 'vitest';
+import { render, screen } from '@testing-library/react';
+import { CardList } from './CardList';
+import type { Character } from '../../types';
+
+describe('CardList', () => {
+  const characters: Character[] = [
+    {
+      id: 1,
+      name: 'Rick Sanchez',
+      gender: 'Male',
+      image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
+    },
+    {
+      id: 2,
+      name: '',
+      gender: '',
+      image: 'https://rickandmortyapi.com/api/character/avatar/2.jpeg',
+    },
+  ];
+
+  it('renders correct number of items', () => {
+    render(<CardList items={characters} />);
+    expect(screen.getAllByRole('listitem')).toHaveLength(2);
+  });
+
+  it('displays character names and gender', () => {
+    render(<CardList items={characters} />);
+    expect(screen.getByText('Rick Sanchez')).toBeInTheDocument();
+    expect(screen.getByText('Male')).toBeInTheDocument();
+  });
+
+  it('handles missing name and gender gracefully', () => {
+    render(<CardList items={characters} />);
+    expect(screen.getAllByRole('heading')).toHaveLength(2);
+    expect(screen.getAllByRole('heading')[1]).toHaveTextContent('');
+    expect(screen.getAllByText('')[1]).toBeInTheDocument();
+  });
```

Comment:
```suggestion
    expect(screen.getAllByText('')[10]).toBeInTheDocument();
```
вот так тоже проходит. так что не понятно, что он проверяет. после того, как написала тест - сломай его, чтоб ы проверить, что он не ложно-положительный (или как там правильно.. короче что он проходит не просто так)